### PR TITLE
Pr/storybook components

### PR DESCRIPTION
**Pull request by khanlifts**

### CODE REVIEW

File: packages/tocco-ui/src/Button/Button.stories.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ storiesOf('Button', module)
           onClick={action('clicked')}
           look={select('look', {'-': null, ..._pick(stylingLook, ['FLAT', 'RAISED'])}) || undefined}
           iconPosition={select('iconPosition', {'-': null, ...stylingPosition}) || undefined}
-          pending={boolean('pending', false) || undefined}
+          pending={boolean('pending', false)}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Button/Button.stories.js]: why did you change that? this will generate
`<Button label="My Button" onClick={clicked} pending={false} />`

but 

`<Button label="My Button" onClick={clicked}  />`
is desired

File: packages/tocco-ui/src/ButtonGroup/ButtonGroup.stories.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+import {storiesOf} from '@storybook/react'
+import {boolean, select, withKnobs} from '@storybook/addon-knobs'
+
+import ButtonGroup from './'
+import Button from '../Button/Button'
+import ButtonLink from '../ButtonLink/ButtonLink'
+import {stylingInk, stylingLook} from '../utilStyles'
+
+storiesOf('Navigation', module)
```
Comment (MethodenMann) [inline packages/tocco-ui/src/ButtonGroup/ButtonGroup.stories.js]: i dont know why but background addon (https://github.com/storybook-eol/addon-backgrounds)
is not working on your branch

File: packages/tocco-ui/src/ButtonGroup/ButtonGroup.stories.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  .addDecorator(withKnobs)
+  .add(
+    'ButtonGroup',
+    () => (
+      <ButtonGroup
+        ink={select('ink', stylingInk)}
+        look={select('look', stylingLook)}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/ButtonGroup/ButtonGroup.stories.js]: please orient yourself by the button example. This look knob does not work properly (wrong default value and you can choose values that are not allow by proptype)

File: packages/tocco-ui/src/Button/Button.stories.js (hunk 1 above)
Comment (MethodenMann) [inline packages/tocco-ui/src/Button/Button.stories.js]: have you rebased with master an implemented new button examples as we discussed? Im missing the round button

File: packages/tocco-ui/src/EditableValue/EditableValue.stories.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+import EditableValue from './'
+import {htmlMarkup} from '../util/htmlMarkup'
+import {SelectStory} from '../Select/Select.stories'
+import {UploadStory} from '../Upload/Upload.stories'
+import {Upload as Raw} from '../Upload/Upload'
+
+const EMPTY_FUNC = () => {}
+
+storiesOf('Edit Data/EditableValue', module)
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/EditableValue.stories.js]: its hard to test the components if you have to edit the value with knobs. This does not simulate the behavior. Can you change that?

File: packages/tocco-ui/src/EditableValue/EditableValue.stories.js [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  )
+  .add(
+    'Single-Select',
+    () => <SelectStory
+      isMulti={false}
+      delay={0}
+    />
+  )
+  .add(
+    'Multi-Select',
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/EditableValue.stories.js]: "info" box of this story is useless since its shows the stories code and proptypes.

File: packages/tocco-ui/src/util/htmlMarkup.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
+export const htmlMarkup = `
```
Comment (MethodenMann) [inline packages/tocco-ui/src/util/htmlMarkup.js]: this file exists is solely for the story? Please name it like that. 
or better... why dont you just move it into the story.

File: packages/tocco-ui/src/Upload/Upload.stories.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
+/* eslint-disable */
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Upload/Upload.stories.js]: why is this need? usually  there is something wrong if you have to disable lint. or at least only disable a single rule

File: packages/tocco-ui/src/Menu/example.js [hunk 8]
```diff
@@ -8,177 +8,176 @@ import Typography from '../Typography'
...
-    <div>
-      {/* start example */}
-      <h3>Menu Items Flat</h3>
-      <Menu.Stack look="flat">
-        <Menu.Item><Button label="<Button>"/></Menu.Item>
-        <Menu.Item><Button ink="primary" label="<Button primary>"/></Menu.Item>
-        <Menu.Item><ButtonLink href="#Menu" label="<ButtonLink>"/></Menu.Item>
-        <Menu.Item><ButtonLink href="#Menu" ink="primary" label="<ButtonLink primary>"/></Menu.Item>
-      </Menu.Stack>
+export const insertMenuContent = () =>
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Menu/example.js]: please move everything relevant into story file.

File: packages/tocco-ui/src/util/htmlMarkup.js (hunk 6 above)
Comment (khanlifts) [reply packages/tocco-ui/src/util/htmlMarkup.js]: I moved it to a separate file because I had the assumption that code should only be written once and then be referenced. I use this markup in two files.

File: packages/tocco-ui/src/ButtonLink/ButtonLink.stories.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import {storiesOf} from '@storybook/react'
+import {withKnobs, boolean, select, text} from '@storybook/addon-knobs'
+import _pick from 'lodash/pick'
+
+import ButtonLink from './'
+import {stylingInk, stylingLook, stylingPosition} from '../utilStyles'
+
+storiesOf('Navigation', module)
```
Comment (MethodenMann) [inline packages/tocco-ui/src/ButtonLink/ButtonLink.stories.js]: General comment:
Before any of your changes the Components description was shown in the addon info view. Probably it is the same problem as with the background addon? Please figure out what change did remove the description
image

File: packages/tocco-ui/src/EditableValue/EditableValue.stories.js [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+  .add(
+    'Phone',
+    () => <EditableValueStory ChildComponent={PhoneEdit} value={'+41761234567'} />,
+    {info: {propTables: [PhoneEdit], propTablesExclude: [EditableValueStory], source: false}}
+  )
+  .add(
+    'Number',
+    () => <EditableValueStory
+      ChildComponent={NumberEdit}
+      value={12345} options={{prePointDigits: 8, postPointDigits: 2}}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/EditableValue.stories.js]: i think all `option` properties can the shown as knobs?
Would be nice to test prePointDigits and so on.

File: packages/tocco-ui/src/EditableValue/EditableValueStory.stories.js [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import React from 'react'
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/EditableValueStory.stories.js]: please move this into the stories file.

File: packages/tocco-ui/src/EditableValue/typeEditors/NumberEdit.js [hunk 12]
```diff
@@ -78,6 +78,10 @@ const NumberEdit = props => {
   )
 }
 
+NumberEdit.contextTypes = {
```
Comment (MethodenMann) [inline packages/tocco-ui/src/EditableValue/typeEditors/NumberEdit.js]: To change this component properties has nothing to do with storybook in the first place. I expect a separate commit for this.

**This change might affect the entity-browser. Please test everything manually and verify that it still works. Of course you also have to remove intl from the option property. It's always very important to see changes like this in a bigger picture.**

Comment (khanlifts) [reply packages/tocco-ui/src/EditableValue/typeEditors/NumberEdit.js]: I tested all three components and they work as intended.

File: packages/tocco-ui/src/FormField/FormField.stories.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    () =>
+      <FormField
+        id="test2"
+        className="row"
+        label={text('Label', 'Label 1')}
+        mandatory={boolean('Mandatory', false)}
+        dirty={boolean('Dirty', true)}
+        touched={boolean('Touched', true)}
+        error={{
+          error1: [<div key="1"><b>error1-1 line1</b><br/>error1-1 line2</div>],
```
Comment (MethodenMann) [inline packages/tocco-ui/src/FormField/FormField.stories.js]: its it possible to set those as knob?

File: packages/tocco-ui/src/Icon/Icon.stories.js [hunk 14]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import {storiesOf} from '@storybook/react'
+import {select, withKnobs} from '@storybook/addon-knobs'
+
+import {Icon} from './Icon'
+
+storiesOf('Icon', module)
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Icon/Icon.stories.js]: why not add knobs instead of 5 stories similar to Button

File: packages/tocco-ui/src/Layout/Layout.stories.js [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+      <Typography.H5>Footer</Typography.H5>
+    </Panel.Footer>
+  </Panel.Wrapper>
+
+storiesOf('Layout', module)
+  .add(
+    'Layout',
+    () =>
+      <div>
+        {/* start example */}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Layout/Layout.stories.js]: not needed. this is for example code

File: packages/tocco-ui/src/Popover/example.js [hunk 16]
```diff
@@ -7,7 +7,7 @@ import Typography from '../Typography'
 import Link from '../Link'
 // real-import:import {Popover} from 'tocco-ui'
 
-const image = <img src="https://picsum.photos/200/200" width="200" height="200"/>
+export const image = <img src="https://picsum.photos/200/200" width="200" height="200"/>
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Popover/example.js]: why do you change a example.js?

File: packages/tocco-ui/src/Layout/example.js [hunk 17]
```diff
@@ -42,7 +42,7 @@ const ExamplePanel3 = () =>
     </Panel.Footer>
   </Panel.Wrapper>
 
-export default () =>
+export const insertLayoutContent = () =>
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Layout/example.js]: example.js should not be changed

File: packages/tocco-ui/src/FormattedValue/FormattedValue.stories.js [hunk 18]
```diff
@@ -1,30 +1,235 @@
...
+  )
+  .add(
+    'Percent',
+    () => <FormattedValue type="percent" value={number('Percent', 65.89)}/>,
+    {info: {propTables: [PercentFormatter], propTablesExclude: [FormattedValue], source: false}}
+  )
+  .add(
+    'Phone',
+    () => <FormattedValue type="phone" value={text('Phone', '+41761234567')}/>,
+    {info: {propTables: [PhoneFormatter], propTablesExclude: [FormattedValue], source: false}}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/FormattedValue/FormattedValue.stories.js]: I think it's problematic to show Props of PhoneFormatter here since PhoneFormatter is a hidden component that is never used directly. why not simple show propTable and Source of Formatted. So basically delete all these info configs. this does also  make the story cleaner (less code)

File: packages/tocco-ui/src/Menu/Menu.stories.js [hunk 19]
```diff
@@ -0,0 +1,189 @@
...
+import {Icon} from '../Icon/Icon'
+import Menu from './'
+import {Menu as Raw} from './Menu'
+
+storiesOf('Menu', module)
+  .add(
+    'Menu',
+    () =>
+      <div>
+        {/* start example */}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Menu/Menu.stories.js]: this code (comment) is showcase only

File: packages/tocco-ui/src/MultiCheckbox/MultiCheckBox.stories.js [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+        />
+        <MultiCheckbox
+          status="checked"
+          onChange={action('box-action')}
+        />
+        <MultiCheckbox
+          status="indeterminate"
+          onChange={action('box-action')}
+        />
+        {/* end example */}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/MultiCheckbox/MultiCheckBox.stories.js]: remove all comments like this

File: packages/tocco-ui/src/MultiCheckbox/MultiCheckBox.stories.js [hunk 21]
```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-console */
```
Comment (MethodenMann) [inline packages/tocco-ui/src/MultiCheckbox/MultiCheckBox.stories.js]: remove all unsued eslint disables

File: packages/tocco-ui/src/Popover/Popover.js [hunk 22]
```diff
@@ -60,7 +60,7 @@ class Popover extends React.Component {
       {this.state.showToolTip && content
       && ReactDOM.createPortal(
         <Popper
-          modifiers={{preventOverflow: {enabled: false}}}
+          modifiers={{preventOverflow: {enabled: false}, hide: {enabled: false}}}
```
Comment (MethodenMann) [inline packages/tocco-ui/src/Popover/Popover.js]: remove

File: packages/tocco-ui/src/Icon/Icon.stories.js (hunk 14 above)
Comment (khanlifts) [reply packages/tocco-ui/src/Icon/Icon.stories.js]: There's more text to support each case this way.

File: packages/tocco-ui/src/Popover/Popover.js (hunk 22 above)
Comment (khanlifts) [reply packages/tocco-ui/src/Popover/Popover.js]: I added this to prevent this console warning:
`preventOverflow` modifier is required by `hide` modifier in order to work, be sure to include it before `hide`!
